src/nvim/api/buffer.c

```diff
@@ -1546,6 +1570,14 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id,
     col2 = 0;
   }
 
+  // Only error out if they try to set end_right_gravity to true without
+  // setting end_col or end_line
+  if (line2 == 0 && col2 == 0 && end_right_gravity) {
+    api_set_error(err, kErrorTypeValidation,
+                  "cannot set end_right_gravity to true "
```

Comment:
"cannot set end_right_gravity" is enough, neither value makes sense if end doesn't exist.